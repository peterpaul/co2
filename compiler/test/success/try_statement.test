

["stdio.h"] int printf(char * format, ...);

class Exception
{
	char * message;
}

class AnException : Exception
{
}

int fun (Exception ex)
{
	if (ex != null)
	{
		throw ex;
	}
}

void loop (Exception ex)
{
	printf ("--------------------------------------------------------------------------------\n");
	try {
		printf ("try\n");
		fun (ex);
	} catch (AnException ex) {
		printf ("catch AnException\n");
		delete ex;
	} catch (Exception ex) {
		printf ("catch Exception\n");
		delete ex;
	}
}

int main(int argc, char ** argv)
{
	loop (null);
	loop (new Exception());
	loop (new AnException());
	return 0;
}