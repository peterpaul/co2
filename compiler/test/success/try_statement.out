--------------------------------------------------------------------------------
try
--------------------------------------------------------------------------------
try
catch Exception
--------------------------------------------------------------------------------
try
catch AnException
